www/lib/searchDescriptor.ts

```diff
@@ -0,0 +1,91 @@
+import { Class, Func, Module, Project, Variable } from "./docs";
+import {
+  mod as generateModuleUrl,
+  project as generateProjectUrl,
+  symbol as generateSymbolUrl,
+} from "./url";
+
+export type SearchDescriptor = {
+  name: string;
+  // TODO: finish this list
+  type?: "project" | "module" | "function" | string;
```

Comment:
```suggestion
  type: "project" | "module" | "function" | string;
```

Reply:
As discussed offline, adding "export" as a potential value will take care of the case I thought needed the nullable value. Fixed in e132778240ed9f18adcc356f9d7a9b934130c880, also making the possible values explicit.